saleor/vendor/models.py

```diff
@@ -0,0 +1,41 @@
+from django.conf import settings
+from django.db import models
+
+# Create your models here.
+from ..warehouse.models import Allocation, Warehouse
+
+
+class Vendor(models.Model):
+    """Model definition for Vendor."""
+
+    # TODO: Define fields here
+    user = models.OneToOneField(
+        settings.AUTH_USER_MODEL,
+        blank=True,
+        null=True,
+        related_name="vendor",
+        on_delete=models.CASCADE,
+    )
+    # here the ideas clash
+    # one django doesn't have onetomany only manytoone
+    # so either I make vendor foreign key in warehouse
+    # or here is a dirty little trick to make manytomany
+    # into one to many
+    warehouse = models.ManyToManyField(Warehouse, related_name="vendors")
+    allocation = models.ManyToManyField(Allocation, related_name="vendors")
```

Comment:
same related name in warehouse and allocation .. Problems above mention comment.

Reply:
The names have been changed although no error was observed due to this during migration.